### ISSUE TITLE
[ENHANCEMENT] Allow automatic offset tool calibration per toolhead

### ISSUE THREAD

**Issue by sassda** [prusa3d/Prusa-Firmware-Buddy#3478]
On a Prusa XL 5th
My printer came with a faulty screw threads for the 2nd extruder, and while I work that out with support, it would be great to be able to do the offset tool calibration only for some toolheads. Specially as this is already possible for things like dock position or other tests, that would allow me to more easily use the machine as a 4th in the meantime. Longer term I can see this being useful if you have a breakdown for a particular extruder and need to recalibrate.
Thanks!

**Comment by reprapkok** [comment]
+1

**Comment by r3Fuze** [comment]
I agree that this is an important feature.

I just swapped the nozzle on a single tool and wanted to re-run the tool offset calibration for that tool, but you can only do all 5 tools at once.

I ended up not doing the calibration because all 5 tools need to be perfectly clean and I couldn't be bothered thoroughly cleaning them all since they keep oozing filament while heated and I don't own a brass brush yet.

**Comment by edsai** [comment]
Would be lovely if this enhancement were made making nozzle changes much quicker on the XL.

**Comment by eltron247** [comment]
+1 
This would be great. No need to go through a full 5 heads if I'm only swapping 1 nozzle.

**Comment by catomann** [comment]
+1

**Comment by ricotrot** [comment]
+1

**Comment by Dycus** [comment]
Please, oh please implement this!
Today I swapped my tool 5 nozzle from 0.4mm to a brand-new 0.6mm. No cleaning or prep was needed at all for this new nozzle because it's already clean. I was hoping I could just go into the menu and hit "calibrate tool 5".

But instead I had to unload filament from all 4 other tools (else it would keep oozing when heated for cleaning), then thoroughly clean each one so they could be unnecessarily re-calibrated, just so I could calibrate the one tool that was changed. Then I had to reload all my filaments. It turned a 10-minute job into a 1-hour job.

I'm definitely going to avoid changing my nozzles as much as possible if that's the workflow every time. :(

**Comment by benjaminbrumbaugh** [comment]
This just got me too. Calibrating all the nozzles when you do a swap means heating up and cleaning every toolhead and going through an extended calibration. A task that should take 30 seconds is taking over a half hour.

**Comment by svake** [comment]
I also would need this feature, because changing only one nozzle (e.g. brass to steel) mean cleaning all nozzles and then calibrating all new. This takes long and if the cleaning is not proper, then I change good values of nozzles that worked so far.

**Comment by bkerler** [comment]
I wrote a PR which is untested right now that should implement this functionality.

**Comment by theLostchemist** [comment]
+1

**Comment by edsai** [comment]
> I wrote a PR which is untested right now that should implement this functionality.

Any news?

**Comment by bkerler** [comment]
Actually there were issues with implementing the PR that need to be resolved first, that's why it is in draft mode currently.

**Comment by edsai** [comment]
This issue is important. Bananas that there has been no movement on it.

**Comment by 3d-gussner** [comment]
I will assign this issue to myself to prevent stale-bot.

**Comment by Dlyfer** [comment]
Its wierd that this isent inplemented from start... so f.. annoying to go thru all of them everytime you change nozzels... hope someone can help.

**Comment by rmpel** [comment]
Same here. I couldn't figure out why one tool gave me "Stuck filament detected" errors, but only during multi-tool prints. Then it hit me; I cannot assume a new nozzle is exactly in the same position as the replaced one. Which is odd, considering the nozzles are mass-produced and should have exact dimensions...
But turns out, it's about 0.2 mm off ...

A calibration of just tool 2 would be fantastic. And it should be simple as well; home the z on tool 1 on which all other offsets are based, home tool 2 (in my case), record the new offset.

**Comment by obs-devs** [comment]
+ 1 on this, had this issue today changing two nozzles on my 5th XL. 
Noticed another issue was also raised recently #5014 - @bkerler was there any progress on you PR?

**Comment by Gik-0101** [comment]
+1

That would be a huge relief and a useful feature!

**Comment by bkerler** [comment]
@obs-devs due to current developments (indx etc.) it doesn't make sense to write a PR that might already be outdated with the next release, so I'm waiting right now to see how they change the code first.

**Comment by rmpel** [comment]
I'm adding this comment to prevent Github from closing the topic automatically. 
I hope that's o.k. to do, because I really really believe in this feature and would not like to see it disappear into the void.

**Comment by rmpel** [comment]
I guess noone else thinks this feature would be a great improvement, so this is the final time I am responding to keep this ticket alive.

**Comment by Gik-0101** [comment]
You aren't the only one who considers this feature extremely useful.
It really should have been implemented a long time ago.

Unfortunately, there has been no reaction from Prusa since the Index launched—
whether regarding firmware bugs or improvements.

**Comment by rmpel** [comment]
yeah, INDX has been taking up practically all Prusa developers their time, I sure hope Prusa will ever get their investment back on that.
